File: SysGestionTurismo/src/main/java/pe/edu/upeu/sysgestionturismo/modelo/Destino.java
package pe.edu.upeu.sysgestionturismo.modelo;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Destino {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idDestino;

    private String nombre;
    private String descripcion;
    private String ubicacion;
}

File: SysTurismoJPC/app/src/main/java/pe/edu/upeu/systurismojpc/modelo/Destino.kt
package pe.edu.upeu.systurismojpc.modelo

data class DestinoDto(
    var idDestino: Long,
    var nombre: String,
    var descripcion: String,
    var ubicacion: String
)

data class DestinoSimpleDto(
    var idDestino: Long
)

data class DestinoResp(
    val idDestino: Long,
    val nombre: String,
    val descripcion: String,
    val ubicacion: String
)

fun DestinoResp.toDto(): DestinoDto {
    return DestinoDto(
        idDestino = this.idDestino,
        nombre = this.nombre,
        descripcion = this.descripcion,
        ubicacion = this.ubicacion
    )
}
